assets/src/models/routePatternData.ts

```diff
@@ -0,0 +1,40 @@
+import { define, Infer, nullable, number, string, type } from "superstruct"
+import { RoutePattern } from "../schedule"
+import { ShapeData, shapeFromData } from "./shapeData"
+
+const DirectionId = define<0 | 1>(
+  "DirectionId",
+  (value) => value === 1 || value === 0
+)
```

Comment:
**thought (non-blocking):** I feel like this is going to be common to a lot of data types that we receive and handle via Superstruct, so it might be nice to start a specific module for those somewhere.